crates/provider/src/kubernetes/namespace.rs

```diff
@@ -0,0 +1,239 @@
+        let scripts_dir = PathBuf::from(format!("{}{}", base_dir_raw, NODE_SCRIPTS_DIR));
+        // NOTE: in native this base path already exist
+        self.filesystem.create_dir_all(&base_dir).await?;
+        try_join!(
+            self.filesystem.create_dir(&config_dir),
+            self.filesystem.create_dir(&data_dir),
+            self.filesystem.create_dir(&scripts_dir),
+        )?;
+
+        // creat k8s pod
+        let manifest = self
+            .client
+            .create_pod(
+                &self.name,
+                &options.name,
+                PodSpec::default(),
+                BTreeMap::from([("foo".to_string(), "bar".to_string())]),
+            )
+            .await
+            .unwrap();
+
+        // store pod manifest
+        self.filesystem
+            .write(
+                PathBuf::from_iter([&base_dir, &PathBuf::from("pod_manifest.yaml")]),
+                serde_yaml::to_string(&manifest).unwrap(),
+            )
+            .await?;
+
+        // create paths
+        let ops_fut: Vec<_> = options
+            .created_paths
+            .into_iter()
+            .map(|created_path| {
+                self.client.pod_exec(
+                    &self.name,
+                    &options.name,
+                    vec!["mkdir", "-p", &created_path.to_string_lossy()]
+                        .into_iter()
+                        .map(|s| s.to_string())
+                        .collect(),
+                )
+            })
+            .collect();
+        // TODO: handle the error conversion correctly
+        try_join_all(ops_fut).await.unwrap();
+
+        // copy injected files
+        let ops_fut: Vec<_> = options
+            .injected_files
+            .iter()
+            .map(|file| {
+                self.client.copy_to_pod(
+                    &self.name,
+                    &options.name,
+                    &file.local_path,
+                    &file.remote_path,
+                    &file.mode,
+                )
+            })
+            .collect();
+        // TODO: handle the error conversion correctly
+        try_join_all(ops_fut).await.unwrap();
+
+        // start process
+        // TODO: handle the error conversion correctly
+        self.client
+            .pod_exec(
+                &self.name,
+                &options.name,
+                vec!["echo", "start", ">", "/tmp/zombiepipe"],
+            )
+            .await
+            .unwrap()
+            .unwrap();
+
+        // create log stream
+        let logs_stream = self
+            .client
+            .create_pod_logs_stream(&options.name, &self.name)
+            .await
+            .unwrap();
+
+        // handle log writing
+        let (logs_tx, rx) = mpsc::channel(10);
+        let log_reading_handle = create_stream_polling_task(logs_stream, logs_tx);
+        let log_writing_handle =
+            create_log_writing_task(rx, self.filesystem.clone(), log_path.clone());
```

Comment:
I don't think we want to write the logs locally, we can always retrieve from the pods (or from grafana if we relay with loki).

Reply:
Removed, make code cleaner, k8s is complex enough without it, can still be used in the future if we need it (from git history)